static/js/buddy_data.js

```diff
@@ -340,6 +330,10 @@ export function get_filtered_and_sorted_user_ids(user_filter_text) {
     return sort_users(user_ids);
 }
 
+export function matches_filter(user_filter_text, user_id) {
+    return get_user_id_list(user_filter_text).includes(user_id);
```

Comment:
I think we can resolve this by making `get_user_id_list` take an optional list of `user_ids`, like `filter_user_ids` does, and only fetching `people.get_active_user_ids` if one was not passed in.

(Or we could skip this refactor, and just add muting logic in `redraw_user`, keeping this a pure function for just the filtering logic, which might be a cleaner abstraction barrier).

Reply:
I understood the performance concern, but not the second solution. `activity.js/redraw_user` was where I had put this logic earlier, and you had suggested that `buddy_list.js/insert_or_move` would be a better place in https://github.com/zulip/zulip/pull/16915#discussion_r608295934.

My reasoning behind doing this refactor (instead of going with `buddy_list.js/insert_or_move`) was that-
- `buddy_list.js` mostly contains code for UI manipulation, as opposed to `buddy_data.js` which handles data.
-  It would be better to have this condition in a single place rather than two.

Should I go ahead with the first solution in your comment?

Reply:
Both solutions seem fine to me but @showell FYI since he may have thoughts.